### PR TITLE
Block access to settings*.php and sites.php

### PR DESCRIPTION
**Pull request by pjcdawkins**
Compare:

https://master-7rqtwti-72zrth3qe73xi.us.platform.sh/sites/default/settings.php
https://pr-43-mwgyqwq-72zrth3qe73xi.us.platform.sh/sites/default/settings.php

https://master-7rqtwti-72zrth3qe73xi.us.platform.sh/sites/default/settings.platformsh.php
https://pr-43-mwgyqwq-72zrth3qe73xi.us.platform.sh/sites/default/settings.platformsh.php

### CODE REVIEW

**Comment by Crell** [comment]
This may be overly aggressive.  Modules can still live in sites/, and some of them do have their own public PHP files for special cases.  (Less so in D8, but I think they still exist.)  I'd rather we block the settings files more surgically than a blanket sites blocker.

**Comment by pjcdawkins** [comment]
Updated, with a more targeted approach.

File: .platform.app.yaml [hunk 1]
```diff
@@ -83,6 +83,12 @@ web:
                 '^/sitemap\.xml$':
                     allow: true
 
+                # Deny direct access to configuration files.
+                '^/sites/sites\.php$':
+                    scripts: false
+                '^/sites/default/settings.*?\.php$':
```
Comment (Crell) [inline .platform.app.yaml]: A multisite may be using `sites/somethingelse/settings.php`.  We should be safe to block `/sites/*/settings.*?.php$`, I think.

Comment (pjcdawkins) [reply .platform.app.yaml]: Agreed - updated

**Pull request merged**